src/dataflow/src/render/debezium.rs

```diff
@@ -173,7 +187,7 @@ struct DebeziumDeduplicationState {
     ///
     /// [`DebeziumEnvelope`] determines whether messages that are not ahead
     /// of the last recorded position will be skipped.
-    last_position_and_offset: Option<(RowCoordinates, Option<i64>)>,
+    last_position_and_offset: HashMap<PartitionId, Option<(RowCoordinates, Option<i64>)>>,
```

Comment:
For my work on trying to persist `DebeziumDeduplicationState` I will probably have to scope the whole of `DebeziumDeduplicationState` to a partition, otherwise it will be hard (impossible, looks like) to restore state, mostly because the number of workers can change. You could keep `DebeziumDeduplicationState` and `should_use_record()` as is, keep a `HashMap<PartitionId, DedebeziumDeduplicationState>` in the operator itself, and just fetch (or create) the right state for the incoming record. WDYT?